pycm/pycm_curve.py

```diff
@@ -258,6 +341,31 @@ def __curve_thresholds_handler__(curve, thresholds):
         curve.thresholds = thresholds
         if isinstance(curve.thresholds, numpy.ndarray):
             curve.thresholds = curve.thresholds.tolist()
+        curve.thresholds = sorted(curve.thresholds)
+
+
+def __curve_data_filter__(curve):
+    """
+    Filter data.
+
+    :param curve: curve
+    :type curve: pycm.Curve object
+    :return: None
+    """
+    none_warning = False
+    for c_index, c in enumerate(curve.classes):
+        data_temp = {curve.plot_x_axis: [], curve.plot_y_axis: []}
+        x_data = curve.data[c][curve.plot_x_axis]
+        y_data = curve.data[c][curve.plot_y_axis]
+        for index, item in enumerate(x_data):
+            if item != "None" and y_data[index] != "None":
+                data_temp[curve.plot_x_axis].append(item)
+                data_temp[curve.plot_y_axis].append(y_data[index])
```

Comment:
`for x, y in zip(x_data, y_data)`

Reply:
Fixed in https://github.com/sepandhaghighi/pycm/pull/467/commits/be9783e7a77569cc525fd9c76a9ab2bfdc5b744d